Autoinst #9216 ke (#827)
SUSE/doc-susemanager: modules/client-configuration/pages/client-automating-preparation.adoc

== Preparation

A configured distribution and an autoinstallation profile is required.

.Procedure: Preparing a Distribution

. Provide the files required to start an installation.
Unpack an installation medium such as a DVD image on your Server.
It contains the Linux kernel, an initrd, and other files required to boot the OS in installation mode.

. In the {productname} {webui}, navigate to menu:System[Autoinstallation > Distributions].

. In the [guimenu]``Autoinstallable Distributions`` dialog, click [guimenu]``Create Distribution``:
* In the [guimenu]``Distribution Label`` field, enter a name to identify your autoinstallable distribution.
* In the [guimenu]``Tree Path`` field, enter the path to an installation tree located on your {productname} Server.
* Select the matching [guimenu]``Base Channel`` mirrored on the {productname} Server.
This base channel must represent the distribution you want to install.
It can be the [guimenu]``Vendor``, [guimenu]``Custom``, or [guimenu]``Cloned Channels``.
* The [guimenu]``Installer Generation`` should also match.
* Optionally, you can specify kernel options which should be added when booting this distribution.

> NOTE: There are multiple places where kernel options can be provided. Only add options here that are generic for the distribution.

Reviewer: ```suggestion
Note that there are multiple places where you can provide kernel options. 
Only add options here that are generic for the distribution.
```

. Click btn:[Create Autoinstallable Distribution].

For more information, see xref:reference:systems/autoinst-distributions.adoc[].

.Procedure: Preparing a Profile

. In the {productname} {webui}, navigate to menu:System[Autoinstallation > Profiles].

. In the [guimenu]``Autoinstallation Profiles`` dialog, add the profile for your  autoinstallation.
It can be an ``AutoYaST`` or ``Kickstart`` profile.

. There are two ways to create profiles:
* Create a ``Kickstart`` profile using a wizard
* Upload an externally created profile (``Kickstart`` or ``AutoYaST``)

For more information about the ``Kickstart`` wizard, see xref:reference:systems/autoinst-profiles.adoc[].

> == Create a Profile

Reviewer 2: Is this section more for "Upload" a profile?

Author: yes, sounds better.  I'll submit the change soon.

> You provide a label for the profile and select an ``Autoinstallation Tree`` (distribution) that you want to use for this profile.

Reviewer: ```suggestion
Profiles require a label, and an ``Autoinstallation Tree`` (distribution).
```

Virtualization type.
// later?  and again kernel options.

> Write, or browse and upload the ``Kickstart`` or ``AutoyaST`` profile.

Reviewer: ```suggestion
You can write your own kickstart or {ay} profile directly in the {webui}, or browse and upload the profile from your local filesystem.
```

> [NOTE]

Reviewer: ```suggestion
```

> ====

Reviewer: ```suggestion
```

AutoYaST is able to dump a profile from an existing installation.

> This can be used as template, but needs some adaptions to be usable from {productname}:

Reviewer: ```suggestion
This can be used as template, but will need to be edited to make it usable by {productname}:
```

> ====

Reviewer: ```suggestion
```

* Change the [literal]``add-on`` section and add {productname} URLs.
These URLs must have a special format:
+

----
http://$redhat_management_server/ks/dist/child/<channel-label>/<distribution-label>
----

* Replace [literal]``<channel-label>`` and [literal]``<distribution-label>`` with the correct labels.

> It often makes sense to use a variable for `distribution-label`.

Reviewer: ```suggestion
You can also use a variable for `distribution-label`.
```

> Ensure that the distribution label corresponds to the Autoinstallable Distribution you have selected.

Reviewer: ```suggestion
Ensure that the distribution label corresponds to the autoinstallable distribution you selected.
```

> You can only configure "child channels" here. These channels must be children of the base channel you selected in the distribution you use for this profile.

Reviewer: ```suggestion
You can only configure child channels in this file. 
The channels must be children of the base channel you selected in the distribution you use for this profile.
```

+
[NOTE]
====
You do not need to specify a base channel.

> You defined the base channel in the distribution already.

Reviewer: ```suggestion
The base channel is defined in the distribution.
```

====

* Register the system after it is installed.
For this step we provide script snippets that can be used.
+

> For Salt managed clients, use the `spacewalk/sle_register_salt` snippet:

Reviewer 2: Hmm, seems I wrote this name in the doc, but named the script different. What should we do now? @juliogonzalez what is the status of re-submitting 4.0.3 ?

> For Salt managed clients, use the `spacewalk/sle_register_salt` snippet:

Reviewer: We can just the change the doc to be the same as the script, can't we?

+
----
<scripts>
  <init-scripts config:type="list">
    $SNIPPET('spacewalk/sle_register_salt')
  </init-scripts>
</scripts>
----
+
For traditional clients, use the `spacewalk/sles_register_script` snippet:
+
----
<scripts>
  <init-scripts config:type="list">
    $SNIPPET('spacewalk/sles_register_script')
  </init-scripts>
</scripts>
----
+
[IMPORTANT]
====

> When registering with Salt, the Salt key must be accepted on the {productname} Server before the autoinstallation procedure could be finished.

Reviewer: ```suggestion
For registering Salt clients, you must accept the Salt key on the {productname} Server before you attempt autoinstallation.
```

====

For more information about autoinstallation profiles, see xref:reference:systems/autoinst-profiles.adoc[].

==== Variables and snippets

> The final state of a profile is composed when a client requests it.

Reviewer: ```suggestion
Profiles are not finalized until they are requested by a client.
```

> This allows you using variables in profiles.

Reviewer: ```suggestion
This allows you to use variables in profiles.
```

In [guimenu]``Profiles`` you can find the [guimenu]``Variables`` where you can define variables.

> Often used variables are:

Reviewer: ```suggestion
Some common variables are:
```

redhat_management_server::

> the server where you register to (automatically set).

Reviewer: ```suggestion
the server {redhat} clients register to (automatically set).
```

org::
the organization ID where this profile is created (automatically set).
registration_key::
the key used in the registration snippets.

> When you specify it, you can set the activation key to be used to register the system.

Reviewer: ```suggestion
By specifying this variable, you can set the activation key to be used to register the system.
```

dont_register::

> if specified the registration will be skipped.

Reviewer: ```suggestion
if specified, the registration will be skipped.
```

allow_config_actions::
if set to ``1``, it will allow traditional configuration management (traditional only).
allow_remote_commands::
if set to ``1``, it will allow traditional remote command execution (traditional only).
dont_disable_automatic_onlineupdate::
if set, the automatic online update will stay enabled (SUSE OSes only).
dont_disable_local_repos::

> if set, local repositories will stay active (not recommended).

Reviewer: The descriptions need capital letters, if you are giving them fullstops ;)

Author: Fair proposal ;) My excuse: the fullstops were already provided before I started editing (IIRC).
